packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/ClusterManagersController.m

```diff
@@ -0,0 +1,157 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "ClusterManagersController.h"
+#import "FLTGoogleMapJSONConversions.h"
+
+@interface FLTClusterManagersController ()
+
+@property(strong, nonatomic) NSMutableDictionary *clusterManagerIdToManagers;
+@property(strong, nonatomic) FlutterMethodChannel *methodChannel;
+@property(weak, nonatomic) GMSMapView *mapView;
+
+@end
+
+@implementation FLTClusterManagersController
+- (instancetype)init:(FlutterMethodChannel *)methodChannel mapView:(GMSMapView *)mapView {
+  self = [super init];
+  if (self) {
+    _methodChannel = methodChannel;
+    _mapView = mapView;
+    _clusterManagerIdToManagers = [[NSMutableDictionary alloc] init];
+  }
+  return self;
+}
+
+- (void)addClusterManagers:(NSArray *)clusterManagersToAdd {
+  for (NSDictionary *clusterManager in clusterManagersToAdd) {
+    NSString *identifier = clusterManager[@"clusterManagerId"];
+    id<GMUClusterAlgorithm> algorithm = [[GMUNonHierarchicalDistanceBasedAlgorithm alloc] init];
+    id<GMUClusterIconGenerator> iconGenerator = [[GMUDefaultClusterIconGenerator alloc] init];
+    id<GMUClusterRenderer> renderer =
+        [[GMUDefaultClusterRenderer alloc] initWithMapView:_mapView
+                                      clusterIconGenerator:iconGenerator];
+    GMUClusterManager *clusterManager = [[GMUClusterManager alloc] initWithMap:_mapView
+                                                                     algorithm:algorithm
+                                                                      renderer:renderer];
+    self.clusterManagerIdToManagers[identifier] = clusterManager;
+  }
+}
+
+- (void)removeClusterManagers:(NSArray *)identifiers {
+  for (NSString *identifier in identifiers) {
+    GMUClusterManager *clusterManager = [self.clusterManagerIdToManagers objectForKey:identifier];
+    if (!clusterManager) {
+      continue;
+    }
+    [clusterManager clearItems];
+    [self.clusterManagerIdToManagers removeObjectForKey:identifier];
+  }
+}
+
+- (void)addItem:(GMSMarker *)marker clusterManagerId:(NSString *)clusterManagerId {
+  GMUClusterManager *clusterManager =
+      [self.clusterManagerIdToManagers objectForKey:clusterManagerId];
+  if (marker && clusterManager != (id)[NSNull null]) {
+    [clusterManager addItem:(id<GMUClusterItem>)marker];
+  }
+}
+
+- (void)removeItem:(GMSMarker *)marker clusterManagerId:(NSArray *)clusterManagerId {
+  GMUClusterManager *clusterManager =
+      [self.clusterManagerIdToManagers objectForKey:clusterManagerId];
+  if (marker && clusterManager != (id)[NSNull null]) {
+    [clusterManager removeItem:(id<GMUClusterItem>)marker];
+  }
+}
```

Comment:
Can these methods be replaced by a single 
` - (GMUClusterManager *)getClusterManagerWithIdentifier:(NSString *)identifier`? Then the user can just call the `removeItem` or addItem` methods on the manager.

Reply:
Refactored the code to use this pattern.